test/integration/versioned_functional_test.go

```diff
@@ -0,0 +1,48 @@
+// +build integration versioned
```

Comment:
I think you might need to make this:
```go
// +build integration
// +build versioned
```

Otherwise these become an OR instead of an AND:
https://golang.org/pkg/go/build/